src/PythonClient/src/quixstreams/dataframes/platforms/quix/config.py

```diff
@@ -0,0 +1,285 @@
+        """
+        Queries for workspace data from the Quix API, regardless of instance cache,
+        and updates instance attributes from query result.
+
+        :param known_workspace_topic: a topic you know to exist in some workspace
+        """
+        # TODO: more error handling with the wrong combo of ws_id and topic
+        if self._workspace_id:
+            ws_data = self.search_for_workspace(workspace_name_or_id=self._workspace_id)
+        else:
+            ws_data = self.search_for_topic_workspace(known_workspace_topic)
+        if not ws_data:
+            raise self.NoWorkspaceFound(
+                "No workspace was found for the given workspace/auth-token/topic combo"
+            )
+        self._workspace_id = ws_data.pop("workspaceId")
+        self._quix_broker_config = ws_data.pop("broker")
+        self._quix_broker_settings = ws_data.pop("brokerSettings")
+        self._workspace_meta = ws_data
+
+    def search_workspace_for_topic(self, workspace_id: str, topic: str):
+        """
+        Search through all the topics in the given workspace id to see if there is a
+        match with the provided topic.
+
+        :param workspace_id: the workspace to search in
+        :param topic: the topic to search for
+        :return: the workspace_id if success, else None
+        """
+        topics = self.api.get_topics(workspace_id=workspace_id)
+        for t in topics:
+            if t["name"] == topic or t["id"] == topic:
+                return workspace_id
+        return None
+
+    def search_for_topic_workspace(self, topic: str):
+        """
+        Find what workspace a topic belongs to.
+        If there is only one workspace altogether, it is assumed to be the workspace.
+        More than one means each workspace will be searched until the first hit.
+
+        :param topic: the topic to search for
+        :return: workspace data dict if topic search success, else None
+        """
+        ws_list = self.api.get_workspaces()
+        if len(ws_list) > 1:
+            if topic is None:
+                raise self.MultipleWorkspaces(
+                    "More than 1 workspace was found, so you must provide a topic name "
+                    "to find the correct workspace"
+                )
+            print(ws_list)
+            for ws in ws_list:
+                if self.search_workspace_for_topic(ws["workspaceId"], topic):
+                    return ws
+            return None
+        else:
+            return ws_list[0]
+
+    def get_workspace_ssl_cert(self, extract_to_folder: Optional[Path] = None) -> str:
+        """
+        Gets and extracts zipped certificate from the API to provided folder.
+        If no path was provided, will dump to /tmp. Expects cert named 'ca.cert'.
+
+        :param extract_to_folder: path to folder to dump zipped cert file to
+        :return: full cert filepath as string
+        """
+        if not extract_to_folder:
```

Comment:
Oh man, good catch with `/tmp`. You can see how much I mess with Windows =)

In terms of using the dir, that function is mostly if a user wanted an easy way to grab the cert themselves for whatever reason; the class actually uses `_set_workspace_cert` under the hood (which calls `get_workspace_ssl_cert`) when you call things like `get_confluent_client_configs`  and it uses/sets the same dir as the C client does (if you didn't hand a path at object init): `Path(getcwd()) / "certificates" / self.workspace_id`. Let me know if you think that's too janky.

Reply:
Ok, that part makes sense to me.
There's one thing that's not clear:

The `_set_workspace_cert()` function contains `if self._workspace_cert_path` clause.
But the function itself is executed only if `not self._workspace_cert_path`.

In this case, will this code ever be executed?:
```
 if self._workspace_cert_path:
            folder = Path(self._workspace_cert_path)
            if folder.name.endswith("cert"):
                folder = folder.parent
  ```

Reply:
Ah yeah, you were right; I meant to have `cfg_out["ssl.ca.location"] = self._set_workspace_cert()` so that if the user provided a cert path at init, it would still generate it correctly (if needed) when getting configs. I think that got lost in translation as I was refactoring, good catch!